pkg/coinbase/staking_operation.go

```diff
@@ -206,50 +206,60 @@ func (c *Client) Wait(ctx context.Context, stakingOperation *StakingOperation, o
 	startTime := time.Now()
 
 	for time.Since(startTime).Seconds() < float64(options.timeoutSeconds) {
-		so, err := c.fetchExternalStakingOperation(ctx, stakingOperation)
-		if err != nil {
-			return stakingOperation, err
+		if err := c.ReloadStakingOperation(ctx, stakingOperation); err != nil {
+			return err
 		}
-		stakingOperation = so
 
 		if stakingOperation.isTerminalState() {
-			return stakingOperation, nil
+			return nil
 		}
 
 		if time.Since(startTime).Seconds() > float64(options.timeoutSeconds) {
-			return stakingOperation, fmt.Errorf("staking operation timed out")
+			return fmt.Errorf("staking operation timed out")
 		}
 
 		time.Sleep(time.Duration(options.intervalSeconds) * time.Second)
 	}
 
-	return stakingOperation, fmt.Errorf("staking operation timed out")
+	return fmt.Errorf("staking operation timed out")
 }
 
-// FetchExternalStakingOperation reloads a staking operation from the API associated
-// with an address.
-func (c *Client) fetchExternalStakingOperation(ctx context.Context, stakingOperation *StakingOperation) (*StakingOperation, error) {
+// ReloadStakingOperation reloads a staking operation from the backend with the latest state.
+// It ensures only newly constructed transactions are added to the staking operation and any existing transactions are untouched.
+func (c *Client) ReloadStakingOperation(ctx context.Context, stakingOperation *StakingOperation) error {
 	so, httpResp, err := c.client.StakeAPI.GetExternalStakingOperation(
 		ctx,
 		stakingOperation.NetworkID(),
 		stakingOperation.AddressID(),
 		stakingOperation.ID(),
 	).Execute()
 	if err != nil {
-		return nil, errors.MapToUserFacing(err, httpResp)
+		return errors.MapToUserFacing(err, httpResp)
 	}
 
 	stakingOperation.model = so
+
 	for _, tx := range so.Transactions {
 		if !stakingOperation.hasTransactionByUnsignedPayload(tx.UnsignedPayload) {
 			newTx, err := newTransactionFromModel(&tx)
 			if err != nil {
-				return nil, err
+				return err
 			}
+
 			stakingOperation.transactions = append(stakingOperation.transactions, newTx)
 		}
 	}
-	return stakingOperation, nil
+	return nil
+}
+
+// FetchStakingOperation fetches a staking operation from the backend given a networkID, addressID, and stakingOperationID.
+func (c *Client) FetchStakingOperation(ctx context.Context, networkID, addressID, stakingOperationID string) (*StakingOperation, error) {
```

Comment:
Is this the same name we use in other SDKs? I thought we included the word external for some reason? (since you can fetch a wallet based op that is slightly different?)

Reply:
Yes. In both [node](https://github.com/coinbase/coinbase-sdk-nodejs/blob/master/src/coinbase/staking_operation.ts#L45) and [ruby](https://github.com/coinbase/coinbase-sdk-ruby/blob/master/lib/coinbase/staking_operation.rb#L248) we seem to call it just `fetch` which takes an optional wallet id as input based on which `GetExternalStakingOperation` and `GetStakingOperation` is called.

Since Go SDK is purely external address based for now, I decided not to introduce the wallet id param and simply call `GetExternalStakingOperation` internally.